app/src/main/java/com/sun/chat_04/ui/signup/SignUpActivity.kt

```diff
@@ -13,32 +13,106 @@ import android.view.View
 import android.view.View.OnClickListener
 import android.widget.DatePicker
 import android.widget.RadioButton
+import android.widget.Toast
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.FirebaseDatabase
 import com.jpardogo.android.googleprogressbar.library.ChromeFloatingCirclesDrawable
+import com.sun.chat_04.R
 import com.sun.chat_04.R.color
 import com.sun.chat_04.R.id
 import com.sun.chat_04.R.layout
 import com.sun.chat_04.data.model.User
-import kotlinx.android.synthetic.main.sign_up_screen.button_signUp
-import kotlinx.android.synthetic.main.sign_up_screen.edit_confirm_password
-import kotlinx.android.synthetic.main.sign_up_screen.edit_email
-import kotlinx.android.synthetic.main.sign_up_screen.edit_fullname
-import kotlinx.android.synthetic.main.sign_up_screen.edit_password
+import com.sun.chat_04.data.remote.UserRemoteDataSource
+import com.sun.chat_04.data.repositories.UserRepository
+import kotlinx.android.synthetic.main.sign_up_screen.buttonSignUp
+import kotlinx.android.synthetic.main.sign_up_screen.editConfirmPassword
+import kotlinx.android.synthetic.main.sign_up_screen.editEmail
+import kotlinx.android.synthetic.main.sign_up_screen.editFullname
+import kotlinx.android.synthetic.main.sign_up_screen.editPassword
 import kotlinx.android.synthetic.main.sign_up_screen.progress
-import kotlinx.android.synthetic.main.sign_up_screen.radio_groups
-import kotlinx.android.synthetic.main.sign_up_screen.text_birthday
+import kotlinx.android.synthetic.main.sign_up_screen.radioGroups
+import kotlinx.android.synthetic.main.sign_up_screen.textBirthday
 import kotlinx.android.synthetic.main.sign_up_screen.toolbar
 import java.util.Calendar
 
-class SignUpActivity : AppCompatActivity(), OnClickListener {
-    private var user: User? = null
+class SignUpActivity : AppCompatActivity(), OnClickListener, SignUpContract.View {
+
+    private var presenter: SignUpPresenter? = null
```

Comment:
```suggestion
    private var presenter: SignUpContract.Presenter? = null
```
khai báo contract thì contract của em mới có giá trị